pkg/services/pluginsintegration/clientmiddleware/team_headers_middleware.go

```diff
@@ -0,0 +1,181 @@
+package clientmiddleware
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/services/contexthandler"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+// NewTeamHeaderMiddleware creates a new plugins.ClientMiddleware that will
+// set headers based on teams user is member of.
+func NewTeamHeadersMiddleware() plugins.ClientMiddleware {
+	return plugins.ClientMiddlewareFunc(func(next plugins.Client) plugins.Client {
+		return &TeamHeadersMiddleware{
+			next: next,
+		}
+	})
+}
+
+type TeamHeadersMiddleware struct {
+	next plugins.Client
+}
+
+type TeamHeadersJSONData struct {
+	TeamHeaders TeamHttpHeaders `json:"teamHeaders"`
```

Comment:
@alexanderzobnin should we just change teamHeaders to teamHttpHeaders?

Reply:
yes, sounds more clear